src/weights.jl

```diff
@@ -18,12 +18,25 @@ macro weights(name)
 end
 
 length(wv::AbstractWeights) = length(wv.values)
-values(wv::AbstractWeights) = wv.values
 sum(wv::AbstractWeights) = wv.sum
 isempty(wv::AbstractWeights) = isempty(wv.values)
 size(wv::AbstractWeights) = size(wv.values)
 
-Base.getindex(wv::AbstractWeights, i) = getindex(wv.values, i)
+Base.convert(::Type{AbstractVector}, wv::AbstractWeights) = wv.values
```

Comment:
Sorry, my suggestion didn't make any sense as `AbstractWeights <: AbstractVector`. So we should only define the `Vector` method.
```suggestion
```